jupyterhub/templates/hub/configmap.yaml

```diff
@@ -133,6 +133,9 @@ data:
     {{- .Values.singleuser.cloudMetadata | toYaml | trimSuffix "\n" | nindent 4 }}
   singleuser.start-timeout: {{ .Values.singleuser.startTimeout | quote }}
   singleuser.image-pull-policy: {{ .Values.singleuser.image.pullPolicy | quote }}
+  {{- if .Values.singleuser.image.pullSecret }}
+  singleuser.image-pull-secret: singleuser-image-credentials
```

Comment:
Oh thinking about it, let us not propagate this mistake in KubeSpawner but instead try to alleviate it by calling this `image-pull-secret-name` instead.